# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreChild.cpp
# Commit: 0868fb4018a7
# Full Hash: 0868fb4018a7ae2002b8f2fcfd88643c2e060f07
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 9c8b6e35bf3f -r 0868fb4018a7 dom/cookiestore/CookieStoreChild.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/cookiestore/CookieStoreChild.cpp	Tue Sep 10 16:31:38 2024 +0000
@@ -0,0 +1,28 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "CookieStore.h"
+#include "CookieStoreChild.h"
+#include "mozilla/ipc/PBackgroundChild.h"
+
+namespace mozilla {
+
+using namespace ipc;
+
+namespace dom {
+
+CookieStoreChild::CookieStoreChild() = default;
+
+CookieStoreChild::~CookieStoreChild() = default;
+
+void CookieStoreChild::Close() {
+  if (CanSend()) {
+    SendClose();
+  }
+}
+
+}  // namespace dom
+}  // namespace mozilla